Vostok.ServiceDiscovery.Abstractions/Models/ReplicaTagsHelpers.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+
+namespace Vostok.ServiceDiscovery.Abstractions.Models
+{
+    public static class ReplicaTagsHelpers

```

Comment:
It feels like `Tag[]` is a poor model choice. It doesn't offer lookup and tells users nothing about key uniqueness constraints. I suggest adding a proper tag collection model instead. Here's something off the top of my head:

```
// case insensitive comparer built-in
public class TagCollection : Dictionary<string, string>, IEnumerable<Tag>
{
    public override string ToString(); // replaces Serialize helper

    public static bool TryParse(string input, out TagCollection collection); // replaces Deserialize helper
}
```

`Distinct` helper wouldn't be needed at all due to Dictionary-based structure.

Reply:
I agree with you, it looks more convenient than the current solution. Let's try to replace `Tags[]` in all places where it can be useful

Reply:
I wanna change `Dictionary` to `IDictionary` because i wanna use custom `Add` method in realization: i wanna rewrite old value with my new value. Is it good that i change default logic of IDictionary interface?

Reply:
About `TryParse`: I have doubts about usability of not success value parsing. In our internal methods i better skip this and return empty TagsCollection (or null), for example: 
```
[CanBeNull]
        private static TagCollection GetPersistentReplicaTags([NotNull] this IReadOnlyDictionary<string, string> properties, string replicaName) 
            => properties.TryGetValue(GetPersistentReplicaTagsPropertyKey(replicaName), out var value) 
                && TagCollection.TryParse(value, out var tagCollection)
                ? tagCollection
                : null;
```
But it can be usefull for users. I am interested in your opinion.

Reply:
done